scripts/getChangelog.py

```diff
@@ -1,90 +1,587 @@
-        print("  " + pr_details["title"] + " (#" + str(pr_details["number"]) + ")")
-        prsSeen.add(pr_number)
-        continue
-    time.sleep(1.5) # Avoid rate limit
-    prs = requests.get("https://api.github.com/search/issues?q=repo:" + REPO + "+type:pr+is:merged+" + sha, headers={'Authorization': 'token ' + TOKEN}).json()
-    if len(prs["items"]) == 0:
-        outputFile.write("Commit," + commit["commit"]["committer"]["date"] + "," + commit["html_url"] + ",\"" + commit["commit"]["message"].splitlines()[0] + "\"," + commit["committer"]["login"] + "\n")
-        print("  [orphan] " + commit["commit"]["message"].splitlines()[0])
-        continue
-    pr_details = prs["items"][0]
-    if pr_details["number"] in prsSeen:
-        print_seen()
-        continue
-    outputFile.write("PR," + pr_details["pull_request"]["merged_at"] + "," + pr_details["html_url"] + ",\"" + pr_details["title"] + "\"," + pr_details["user"]["login"] + "\n")
-    print("  " + pr_details["title"] + " (#" + str(pr_details["number"]) + ")")
-    prsSeen.add(pr_details["number"])
-outputFile.close()
+# TODO:
+# Make it possible to run the script through workflow file, completely headless, with only base_ref and head_ref as input
+# Make it possible to run the part of the script that creates a list of PRs that need review replies in a workflow file (which posts a comment for each of the relevant PRs/issues when creating a release)
+# i.e. define two headless states: "get_changelog_CSV" and "get_PRs_that_need_review_replies"
+# However, the former should already be possibe by running the script with only base_ref and head_ref as input
+
+try:
+  from graphqlclient import GraphQLClient
+  import json
+  import time
+  from datetime import datetime
+  import csv
+  from collections import OrderedDict
+  import sys
+  import os
+  import getpass
+  import threading
+  import urllib.error
+  import textwrap
+except ModuleNotFoundError as e:
+  print(f"The '{e.name}' module is not installed. Please install it using 'pip install {e.name}' and try again.")
+  exit()
+
+# Define variables
+owner = os.getenv('OWNER', "AntennaPod") # The owner (organisation or user) of the repository
+repo = os.getenv('REPO', "AntennaPod") # The repository name
+token = os.getenv('GITHUB_API_TOKEN') # The GitHub API token
+base_ref = os.getenv('BASE') # The base reference (release code or branch); point of reference
+head_ref = os.getenv('HEAD') # The head reference (release code or branch); environment containing the changes
+needs_review_reply_label = os.getenv('NEED_REVIEW_REPLY_LABEL', "Needs: Review reply") # The name or ID of the label that indicates that we should reply to a (Google Play) review when the issue or PR that has this label iss addressed or released
+client = GraphQLClient('https://api.github.com/graphql')
+
+# Function: Handle exceptions
+def handle_exception(error_message: str, data: dict, error_introduction="Error"):
```

Comment:
This method says it wants a `str` but it is called with an Exception object. The name `handle` is a bit ambiguous. I would rather call it `print_exception` or something